runtime/parachains/src/disputes/slashing.rs

```diff
@@ -228,18 +230,40 @@ where
 		candidate_hash: CandidateHash,
 		kind: SlashingOffenceKind,
 		losers: impl IntoIterator<Item = ValidatorIndex>,
+		backers: impl IntoIterator<Item = ValidatorIndex>,
 	) {
-		let losers: Vec<ValidatorIndex> = losers.into_iter().collect();
-		if losers.is_empty() {
-			// Nothing to do
+		// sanity check for the current implementation
+		if kind == SlashingOffenceKind::AgainstValid {
+			debug_assert!(false, "should only slash ForInvalid disputes");
+			return
+		}
+		let backers: Vec<ValidatorIndex> = backers.into_iter().collect();
+		debug_assert!(
+			{
+				let losers_index: BTreeSet<_> = losers.into_iter().collect();
+				backers.iter().all(|i| losers_index.contains(i))
```

Comment:
I realized this assumption is not correct. We'll need to slash an intersection of losers and backers as this function can be called multiple times per dispute.